Rework radio channel names

halworsen opened the pull request:
:cl: Atebite
rscadd: Telecommunication servers now have tagging rules which determine the radio channel name and color for a given frequency
/:cl:

Found it annoying that I couldn't set up a new radio channel with a proper name in-round. Now you can do it with tagging rules on the comms servers. Also lets you extend the frequency "range" for the default channels, or swap them around if you like.

![](https://i.imgur.com/LuBzKjK.png)

halworsen committed:
Radio signal tagging

halworsen committed:
Refresh server UI on tag rule add fail

halworsen committed:
Make radio channel color depend purely on channel name

halworsen committed:
Make server preset tags consistent with the radiochannels list

halworsen committed:
Fix antag broadcasts

halworsen committed:
Use channel_tag for signal data

tag was already being used (oops)

halworsen committed:
Channel tag all radio signals as '#unkn' by default

Mostly to display antag frequencies as unknown

PsiOmegaDelta commented on code/game/machinery/telecomms/broadcaster.dm:
```diff
@@ -219,12 +219,15 @@ var/message_delay = 0 // To make sure restarting the recentmessages list is kept
 						var/vmask, var/vmessage, var/obj/item/device/radio/radio,
 						var/message, var/name, var/job, var/realname, var/vname,
-						var/data, var/compression, var/list/level, var/freq, var/verbage = "says", var/datum/language/speaking = null)
+						var/data, var/compression, var/list/level, var/freq, var/channel_tag, var/verbage = "says", var/datum/language/speaking = null)
```
I'm not sure it's a wise idea to put new arguments anywhere but last, in particular given the mess radio code is.

halworsen replied:
Changed in 84d4e2d

Sbotkin commented:
What about colors? I don't know how difficult it is to implement but would be neat!

halworsen committed:
Make channel tag last argument in broadcast procs

Zero-Bits commented:
This is pretty cool for the 2 people who know telcomms. It's got my +1

halworsen committed:
Radio message coloring

Squashed commits:

[e468e574aa] Add color selection for channel tags

Doesn't use them yet. In a bit of a hurry and won't have access to my home computer for a bit so I'll finish and squash later

comma commented:
Looks like you changed the globals file, gotta update the hash in testing scripts

Hubblenaut commented:
Looking forward to Engineers changing the name of the Security channel to 'Shitcurity'.
:+1:

Sbotkin commented:
@Hubblenaut I'll make it brown :+1:
